airflow/dags/gtfs_static_downloader.py

```diff
@@ -0,0 +1,113 @@
+"""
+Download the state of CA GTFS files, async version
+"""
+import intake
+import requests
+import logging
+import zipfile
+import io
+import pathlib
+import datetime
+from airflow import DAG
+from airflow.operators.python_operator import PythonOperator
+
+# TODO: Fix to Pathlib
+catalog = intake.open_catalog("./dags/catalogs/catalog.yml")
+
+
+def make_gtfs_list():
+    """
+    Read in a list of GTFS urls
+    from the main db
+    plus metadata
+    """
+    df = catalog.official_list(
+        csv_kwargs={
+            "usecols": ["ITP_ID", "GTFS", "Agency Name"],
+            "dtype": {"ITP_ID": "float64"},
+        }
+    ).read()
+    # TODO: Figure out what to do with Metro
+    # For now, we just take the bus.
+
+    # TODO: Replace URLs with Zip ones.
+    # For now we filter, and then remove ones that don't contain
+    # zip filters
+    df = df[(df.GTFS.str.contains("zip")) & (df.GTFS.notnull())]
+    return df
+
+
+def clean_urls(url):
+    """
+    take the list of urls, clean as needed
+    """
+    # LA Metro split requires lstrip
+    return url
+
+
+def download_url(**kwargs):
+    run_time = kwargs["execution_date"]
+    url = kwargs["params"]["url"]
+    itp_id = kwargs["params"]["itp_id"]
+    headers = {
+        "User-Agent": (
+            "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_4)"
+            "AppleWebKit/537.36 (KHTML, like Gecko)"
+            "Chrome/83.0.4103.97 Safari/537.36"
+        )
+    }
+    try:
+        r = requests.get(url, headers=headers)
+        r.raise_for_status()
+    except requests.exceptions.HTTPError as err:
+        logging.warning(f"No feed found for {url}, {err}")
+    try:
+        z = zipfile.ZipFile(io.BytesIO(r.content))
+        # replace here with s3fs
+        pathlib.Path(f"/tmp/gtfs-data/{run_time}/{itp_id}").mkdir(
```

Comment:
would be great to have things like paths as params somewhere rather than embedded as magic #s

Reply:
yeah... perhaps something to store in the file? let's chat today.